src/Types/AsciiStringType.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Types;
+
+use Doctrine\DBAL\ParameterType;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+
+class AsciiStringType extends StringType
```

Comment:
@morozov I see we never use `final` for types, but maybe we should start?

Reply:
Absolutely. The requirement for the return type declaration is enforced by the coding standard only in `4.x` but the use of final classes isn't enforced even there. It might be a good topic to explore further at the standard level.